src/driver/src/DefaultDriver.cpp

```diff
@@ -55,6 +66,107 @@ YAML::Node loadConfig(const std::string& source,
     }
 }
 
+YAML::Node parseExternal(const YAML::Node& external,
+                         YamlParameters& params,
+                         const fs::path& phaseConfig) {
+    int keysSeen = 1;
+
+    fs::path path(external["Path"].as<std::string>());
+    path = fs::absolute(phaseConfig / path);
+
+    if (!fs::is_regular_file(path)) {
+        auto os = std::ostringstream();
+        os << "Invalid path to external PhaseConfig: " << path
+           << ". Please ensure your workload file is placed in 'workloads/[subdirectory]/' and the "
+              "'Path' parameter is relative to the 'phases/' directory";
+        throw InvalidConfigurationException(os.str());
+    }
+
+    auto replacement = loadConfig(path.string());
+
+    if (external["Parameters"]) {
+        keysSeen++;
+        auto newParams = external["Parameters"].as<YamlParameters>();
+        params.insert(newParams.begin(), newParams.end());
+    }
+
+    if (external["Key"]) {
+        keysSeen++;
+        const auto key = external["Key"].as<std::string>();
+        if (!replacement[key]) {
+            auto os = std::ostringstream();
+            os << "Could not find top-level key: " << key << " in phase config YAML file: " << path;
+            throw InvalidConfigurationException(os.str());
+        }
+        replacement = replacement[key];
+    }
+
+    if (external.size() != keysSeen) {
+        auto os = std::ostringstream();
+        os << "Invalid keys for 'External'. Please set 'Path' and if any, 'Parameters' in the YAML "
+              "file: "
+           << path << " with the following content: " << YAML::Dump(external);
+        throw InvalidConfigurationException(os.str());
+    }
+
+    return recursiveParse(replacement, params, phaseConfig);
+}
+
+YAML::Node replaceParam(const YAML::Node& input, YamlParameters& params) {
+    if (!input["Name"] || !input["Default"]) {
+        auto os = std::ostringstream();
+        os << "Invalid keys for '^Parameter', please set 'Name' and 'Default' in following node"
+           << YAML::Dump(input);
+        throw InvalidConfigurationException(os.str());
+    }
+
+    auto name = input["Name"].as<std::string>();
+    // The default value is mandatory.
+    auto defaultVal = input["Default"];
+
+    // Nested params are ignored for simplicity.
+    if (auto paramVal = params.find(name); paramVal != params.end()) {
+        return paramVal->second;
+    } else {
+        return input["Default"];
+    }
+}
+
+YAML::Node recursiveParse(const YAML::Node& node,
```

Comment:
I've run into weird problems trying to pass `const YAML::Node&`. It's not well-documented but `YAML::Node`s are basically ref-counted so they're cheap to pass around. Please change this and others to just `YAML::Node`.